### PR TITLE
CAMEL-21309: camel-cxf - Force using sync client when using tracing/opentelemetry as otherwise spans are not working correctly.

### PR DESCRIPTION
**Pull request by davsclaus**
# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java
Comment (johnpoth) [inline components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java:1]: It seems `exchange.getProperty(ACTIVE_SPAN)` always returns null. 

`exchange.getProperty(ExchangePropertyKey.ACTIVE_SPAN)` seems to be what we need

File: components/camel-cxf/camel-cxf-soap/src/main/java/org/apache/camel/component/cxf/jaxws/CxfProducer.java
Comment (johnpoth) [inline components/camel-cxf/camel-cxf-soap/src/main/java/org/apache/camel/component/cxf/jaxws/CxfProducer.java:1]: Same, It seems `exchange.getProperty(ACTIVE_SPAN)` always returns null. 

`exchange.getProperty(ExchangePropertyKey.ACTIVE_SPAN)` seems to be what we need

**Review by johnpoth — CHANGES_REQUESTED** [review]

File: components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java
Comment (davsclaus) [reply components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java:1]: It was lacking a mapping
https://github.com/apache/camel/commit/a290693bd4aadcf188b1c0947a531a26e6c1f73c

But we can optimize to use the key constant as suggested

Comment (davsclaus) [reply components/camel-cxf/camel-cxf-rest/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java:1]: ok changed

**Review by johnpoth — APPROVED** [review]

**Comment by johnpoth** [comment]
LGTM !

**Pull request merged**